hazelcast/src/main/java/com/hazelcast/cache/ICache.java

```diff
@@ -944,4 +944,12 @@
      */
     CacheStatistics getLocalCacheStatistics();
 
+    /**
+     * Opens cache if available (not destroyed).
+     * If cache is already open, does nothing.
+     *
+     * @throws java.lang.IllegalStateException if cache is already destroyed
+     */
+    void open();
```

Comment:
But in that case, since there is no relation between `ICache` and `ICacheInternal` and there will be a cast (in `AbstractHazelcastCacheManager`) to `ICacheInternal` via `instanceof` or forcefully which doesn't not seems good to me.